src/HIE/Bios/Cradle.hs

```diff
@@ -493,37 +527,52 @@ withCabalWrapperTool (mbGhc, ghcArgs) wdir = do
 -- | Given the root directory, get the build dir we are using for cabal
 -- In the `hie-bios` cache directory
 cabalBuildDir :: FilePath -> IO FilePath
-cabalBuildDir work_dir = do
-  abs_work_dir <- makeAbsolute work_dir
+cabalBuildDir workDir = do
+  abs_work_dir <- makeAbsolute workDir
   let dirHash = show (fingerprintString abs_work_dir)
-  getCacheDir ("dist-"<>filter (not . isSpace) (takeBaseName abs_work_dir)<>"-"<>dirHash)
+  getCacheDir ("dist-" <> filter (not . isSpace) (takeBaseName abs_work_dir)<>"-"<>dirHash)
+
+cabalGHCDir :: FilePath -> IO (CradleLoadResult (FilePath, FilePath))
+cabalGHCDir workDir = do
+  libdirLoadResult <- readProcessWithCwd workDir  "cabal" ["exec", "-v0", "--", "ghc", "--print-libdir"] ""
+  libdirLoadResult `bindIO` \libdir -> do
+    exePathLoadResult <- readProcessWithCwd workDir  "cabal" ["exec", "-v0", "--", "ghc", "-e", "putStrLn =<< System.Environment.getExecutablePath"] ""
```

Comment:
This actually needs to be:

```
do e <- System.Environment.getExecutablePath ; System.IO.putStr e
```

in case someone has ImplicitPrelude disables globally/locally.

See https://github.com/haskell/haskell-language-server/pull/2711